support/cas-server-support-person-directory/src/main/java/org/apereo/cas/config/CasPersonDirectoryConfiguration.java

```diff
@@ -307,7 +333,9 @@ public void configureAttributeRepositoryPlan(final PersonDirectoryAttributeRepos
 
         @Override
         public void configureAttributeRepositoryPlan(final PersonDirectoryAttributeRepositoryPlan plan) {
-            plan.registerAttributeRepositories(ldapAttributeRepositories());
+            plan.registerAttributeRepositories(ldapAttributeRepositories(
+                personDirectoryAttributeRepositoryPlanConfigurerListConnectionFactoryListBean()
+            ));
         }
```

Comment:
This is not a good pattern. Here's what you should do instead:

- Remove the `PersonDirectoryAttributeRepositoryPlanConfigurer` from the class
- Create a Bean of type `PersonDirectoryAttributeRepositoryPlanConfigurer`. Mark as conditional.
- Inject and qualify `personDirectoryAttributeRepositoryPlanConfigurerListConnectionFactoryListBean` into that bean.
- Register the ldap attribute repository into the plan as usual.

Reply:
Do you want me to refactor like this?
e007c63ec51b55d019fbba7c9ef9cbba151a5430

BTW I found that I have to create a named class to make check style to pass (lambda too long), I took this chance to put the `connectionFactoryList` into the class. Therefore no more `ListFactoryList` needed.